phoenix-core/src/main/java/org/apache/phoenix/iterate/RegionScannerFactory.java

```diff
@@ -215,7 +223,9 @@ public boolean nextRaw(List<Cell> result) throws IOException {
           if (result.size() == 0) {
             return next;
           }
-          if (ScanUtil.isLocalIndex(scan) && !ScanUtil.isAnalyzeTable(scan)) {
+          if ((ScanUtil.isLocalIndex(scan)
+                  || ScanUtil.isUncoveredGlobalIndex(scan))
```

Comment:
You seem to have this in Line 137 and in GroupedAggregateRegion... as well. I think it is better to have a ScanUtil.isLocalOrUncoveredGlobalIndex

Reply:
I will add it